src/test/java/org/opentripplanner/generate/doc/RoutingModeDocTest.java

```diff
@@ -0,0 +1,72 @@
+package org.opentripplanner.generate.doc;
+
+import static org.opentripplanner.framework.io.FileUtils.assertFileEquals;
+import static org.opentripplanner.framework.io.FileUtils.readFile;
+import static org.opentripplanner.framework.io.FileUtils.writeFile;
+import static org.opentripplanner.generate.doc.framework.DocsTestConstants.DOCS_ROOT;
+import static org.opentripplanner.generate.doc.framework.DocsTestConstants.TEMPLATE_ROOT;
+import static org.opentripplanner.generate.doc.framework.TemplateUtil.replaceSection;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Comparator;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.generate.doc.framework.DocBuilder;
+import org.opentripplanner.generate.doc.framework.GeneratesDocumentation;
+import org.opentripplanner.routing.api.request.StreetMode;
+import org.opentripplanner.transit.model.basic.TransitMode;
+
+@GeneratesDocumentation
+public class RoutingModeDocTest {
+
+  private static final File TEMPLATE = new File(TEMPLATE_ROOT, "RoutingModes.md");
+  private static final File OUT_FILE = new File(DOCS_ROOT, "RoutingModes.md");
+
+  @Test
+  public void updateDocs() {
+    // Read and close inout file (same as output file)
+    String doc = readFile(TEMPLATE);
+    String original = readFile(OUT_FILE);
+
+    var streetBuilder = new DocBuilder();
+
+    // Street modes
+    streetBuilder.header(2, "Street modes", "Street modes");
+    // TODO: Shouln't typedescription be static, as it is for the whole of StreetMode?
+    // Or add another field to the DocumentedEnum that would describe the enum as a whole?
```

Comment:
I agree this is a bit awkward, but I want the `DocumentedEnum` to declare the "contract". Unfortunately there is no way to enforce the implementation of static method, so the method is declared as a member method.
```suggestion
```

Feel free to make a comment about this in the JavaDoc on the `DocumentedEnum#typeDescription()` if you want.